src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -4158,11 +4119,29 @@ grpc_error* CallData::ApplyServiceConfigToCallLocked(
   return GRPC_ERROR_NONE;
 }
 
-void CallData::MaybeInvokeConfigSelectorCommitCallback() {
-  if (on_call_committed_ != nullptr) {
-    on_call_committed_();
-    on_call_committed_ = nullptr;
+void CallData::RecvInitialMetadataReadyForConfigSelectorCommitCallback(
+    void* arg, grpc_error* error) {
+  auto* self = static_cast<CallData*>(arg);
+  if (self->on_call_committed_ != nullptr) {
+    self->on_call_committed_();
+    self->on_call_committed_ = nullptr;
   }
+  // Chain to original callback.
+  Closure::Run(DEBUG_LOCATION, self->original_recv_initial_metadata_ready_,
```

Comment:
Yeah, this is what I was asking about re work that is in this PR that wasn't present pre-split and doesn't seem directly related to the split. Can you explain?